src/segment-loader.js

```diff
@@ -1203,6 +1427,16 @@ export default class SegmentLoader extends videojs.EventTarget {
     this.pendingSegment_ = null;
     this.recordThroughput_(segmentInfo);
     this.addSegmentMetadataCue_(segmentInfo);
+    if (!this.playlist_.attributes.BANDWIDTH) {
+      if (!this.nextSegmentSize_) {
+        this.nextSegmentSize_ = segmentInfo.byteLength;
+      } else {
+        this.nextSegmentSize_ -= this.nextSegmentSize_ /
+          this.segmentMetadataTrack_.cues.length;
+        this.nextSegmentSize_ += segmentInfo.byteLength /
```

Comment:
I'm not sure what we are trying to do in these two lines 😕

Reply:
Added comment that it's calculating the moving average (I know, it's hard to understand when you first see it)